lib/core.js

```diff
@@ -130,6 +130,15 @@ function generate(opts) {
                 // What viewports do you care about?
                 width: dimensions.width,   // viewport width
                 height: dimensions.height  // viewport height
+            }, function(err, stdout) {
```

Comment:
penthouse is promisified by `bluebird` so you either need to use the regular penthouse function and return a promise:
```javascript
return new Promise(function (resolve, reject) {
    penthouse({
        url: file.normalizePath(opts.url),
        css: csspath,
        // What viewports do you care about?
        width: dimensions.width,   // viewport width
        height: dimensions.height  // viewport height
    }, function(err, stdout) {
        // explicit close stdout streams of child processes on exit (delegation from penthouse)
        // see node api doc #child_process_event_exit
        //err.debug is also given, please forward it to console.log if a debug flag was set ...
        var childProcessStdout = (err || {}).stdout || stdout;
        if(typeof childProcessStdout.destroy === 'function'){
            childProcessStdout.destroy();
        }

        if (err) {
            return reject(err);
        } else {
            return resolve(stdout);
        }
    });
});
```
or you'll use `penthouseAsync(...).then(...)`